finalize-commands-20260702.sh

```
@@ -0,0 +1,81 @@
+  status=$(ccall $game 'status()(uint8)')
+  resolvedAt=$(ccall $game 'resolvedAt()(uint64)' | awk '{print $1}')
+  if [ "$status" != "2" ]; then echo "    dispute game $game not resolved DEFENDER_WINS yet (status=$status) — skip"; return; fi
+  local maturity finality proofReady gameReady ready now
+  maturity=$(ccall $portal 'proofMaturityDelaySeconds()(uint256)' | awk '{print $1}')
+  finality=$(ccall $portal 'disputeGameFinalityDelaySeconds()(uint256)' | awk '{print $1}')
+  proofReady=$((provenTs + maturity)); gameReady=$((resolvedAt + finality))
+  ready=$(( (proofReady > gameReady ? proofReady : gameReady) + 1 ))
+  now=$(date +%s)
+  if [ "$now" -lt "$ready" ]; then
+    echo "    finalizable at $(date -u -r $ready '+%b %d %H:%M:%S UTC') — skip"; return; fi
+  if ! env -u ETH_KEYSTORE -u ETH_FROM cast call $portal $cd_hex --from $PROVER --rpc-url "$RPC" > /dev/null; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)